package/upgrade/upgrade_node.sh

```diff
@@ -244,6 +244,57 @@ wait_longhorn_engines() {
     done
 }
 
+patch_logging_event_audit()
+{
+  # enabling logging, audit, event when upgrading from v1.0.3 to 1.1.0
+  # should happen before RKE2 is patched
+  # note: the host '/' is mapped to pod '/host/', refer: pkg/controller/master/upgrade/common.go applyNodeJob
+
+  # get UPGRADE_PREVIOUS_VERSION
+  detect_upgrade
+  echo "The current version is $UPGRADE_PREVIOUS_VERSION, will check Logging Event Audit upgrade node option"
+
+  if test "$UPGRADE_PREVIOUS_VERSION" = "v1.0.3"; then
+    echo "Patch kube-audit policy file"
+    # this file should be there in each NODE
+    # keep syncing with harvester/harvester-installer/pkg/config/templates/rke2-92-harvester-kube-audit-policy.yaml
+
+    ls /host/etc/rancher/rke2/config.yaml.d/
+
+    KUBE_AUDIT_POLICY_FILE=/host/etc/rancher/rke2/config.yaml.d/92-harvester-kube-audit-policy.yaml
+    echo "Create new file $KUBE_AUDIT_POLICY_FILE"
+
+    cat > $KUBE_AUDIT_POLICY_FILE << 'EOF'
+apiVersion: audit.k8s.io/v1
+kind: Policy
+omitStages:
+  - "ResponseStarted"
+  - "ResponseComplete"
+rules:
+  # Any include/exclude rules are added here
+
+  # A catch-all rule to log all other (create/delete/patch) requests at the Metadata level
+  - level: Metadata
+    verbs: ["create", "delete", "patch"]
+    omitStages:
+      - "ResponseStarted"
+      - "ResponseComplete"
+EOF
+
+    # it means the NODE role is rke2-server when 90-harvester-server.yaml exists, patch it
+    RKE2_SERVER_CONFIG_FILE=/host/etc/rancher/rke2/config.yaml.d/90-harvester-server.yaml
+    if [ -f "$RKE2_SERVER_CONFIG_FILE" ]; then
+      echo "Patch rke2 server config file with audit-policy-file parameter"
+      echo "audit-policy-file: $KUBE_AUDIT_POLICY_FILE" >> $RKE2_SERVER_CONFIG_FILE
```

Comment:
`KUBE_AUDIT_POLICY_FILE` variable is `/host/etc/rancher/rke2/config.yaml.d/92-harvester-kube-audit-policy.yaml`.
This will cause the rke2-server to fail to start. We need to remove `/host` part here

Reply:
And this line is not persistent between reboots because `99_custom.yaml` contains generation of the `90-harvester-server.yaml` file.

Reply:
Good catch, thanks.

I am working to patch 99_custom.yaml.

Reply:
The 99_custom.yaml is patched via package/upgrade/patch_99_custom.sh

@bk201  please take a further look, thanks.

Reply:
Getting the error:

```
Patch rke2 server config file with audit-policy-file parameter
+ echo 'Patch rke2 server config file with audit-policy-file parameter'
+ echo 'audit-policy-file: /etc/rancher/rke2/config.yaml.d/92-harvester-kube-audit-policy.yaml'
+ source /usr/local/bin/patch_99_custom.sh
/usr/local/bin/upgrade_node.sh: line 303: /usr/local/bin/patch_99_custom.sh: No such file or directory
```
The file needs to be packaged in the [Dockerfile.](https://github.com/harvester/harvester/blob/master/package/upgrade/Dockerfile)

Reply:
added in latest commit